lib/elixir/test/elixir/macro_test.exs

```diff
@@ -1246,7 +1246,13 @@ defmodule MacroTest do
     assert Macro.quoted_literal?(quote(do: {"foo", 1}))
     assert Macro.quoted_literal?(quote(do: %{foo: "bar"}))
     assert Macro.quoted_literal?(quote(do: %URI{path: "/"}))
+    assert Macro.quoted_literal?(quote(do: <<1, "foo", "bar"::utf16>>))
+    assert Macro.quoted_literal?(quote(do: <<1000::size(8)-unit(4)>>))
+    assert Macro.quoted_literal?(quote(do: <<1000::8*4>>))
+    assert Macro.quoted_literal?(quote(do: <<102::unsigned-big-integer-size(8)>>))
     refute Macro.quoted_literal?(quote(do: {"foo", var}))
+    refute Macro.quoted_literal?(quote(do: <<"foo"::binary-size(name_size)>>))
+    refute Macro.quoted_literal?(quote(do: <<102, rest::binary>>))
   end
```

Comment:
Can we add tests for `size()` with a non-literal size but without any other modifier (`<<"foo"::size(name_size)>>` or something) as well?

Reply:
Done in [f30dfe8](https://github.com/elixir-lang/elixir/pull/12270/commits/f30dfe8ae77e95f4cd39ef4a85f20348cc95fe56)